spec/migrations/add_from_template_permissions_to_roles_with_add_permissions_spec.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+require Rails.root.join("db/migrate/20251113124140_add_from_template_permissions_to_roles_with_add_permissions")
+
+RSpec.describe AddFromTemplatePermissionsToRolesWithAddPermissions, type: :model do
+  # Silencing migration logs, since we are not interested in that during testing
+  subject(:run_migration) { ActiveRecord::Migration.suppress_messages { described_class.migrate(:up) } }
+
+  shared_examples_for "not changing permissions" do
+    it "is not changed" do
+      expect { run_migration }.not_to change { role.reload.permissions }
+    end
+
+    it "does not add any new permissions" do
+      expect { run_migration }.not_to change(RolePermission, :count)
+    end
+  end
+
+  shared_examples_for "migration is idempotent" do
+    context "when the migration is run twice" do
+      before { run_migration }
+
+      it_behaves_like "not changing permissions"
+    end
+  end
+
+  shared_examples_for "adding permissions" do |new_permissions|
+    it "adds the #{new_permissions} permissions for the role" do
+      expect { run_migration }.to change { role.reload.permissions }
+        .from(match_array(permissions))
+        .to match_array(permissions + new_permissions)
```

Comment:
Just caught my eye, probably better for consistency to use brackets for both:
```suggestion
        .from(match_array(permissions))
        .to(match_array(permissions + new_permissions))
```